### PR TITLE
[WIP] Add notebook fixture

### PR DESCRIPTION
**Pull request by rohitsanj**
Ref #5 

Created a `execute_cell` decorator that can be used as follows:

```python
 from testbook import execute_cell

 @execute_cell('path/to/notebook.ipynb', 3)
 def test_foo(executed_cell):
     pass
```
As of now, it just returns the `dict` containing the executed cell. I will extend this to be a wrapper object on which we can perform test assertions. 

cc @MSeal 

EDIT: https://github.com/nteract/testbook/pull/10#issuecomment-604007498

### CODE REVIEW

File: testbook/execute.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import pytest
+import nbformat
+from testbook.testbook.client import TestbookNotebookClient
+
+
+def execute_cell(notebook_filename, cell_index):
+    def decorator(func):
```
Comment (MSeal) [inline testbook/execute.py]: We may want to use `@functools.wraps(func)` https://stackoverflow.com/questions/147816/preserving-signatures-of-decorated-functions to preserve function signatures.

File: testbook/client.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+from nbclient.exceptions import CellExecutionError
+
+
+class TestbookNotebookClient(NotebookClient):
+    """
+    Module containing a  that executes the code cells
+    and updates outputs
+    """
+
+    def __init__(self, nb, km=None, **kw):
```
Comment (MSeal) [inline testbook/client.py]: Don't actually need this if it's just calling super

File: testbook/client.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        with self.setup_kernel(**kwargs):
+            self.log.info("Executing notebook with kernel: %s" % self.kernel_name)
+            self.testbook_execute_cells()
+            info_msg = self._wait_for_reply(self.kc.kernel_info())
+            self.nb.metadata['language_info'] = info_msg['content']['language_info']
+            self.set_widgets_metadata()
+
+        return self.nb
+
+    def testbook_execute_cell(
```
Comment (MSeal) [inline testbook/client.py]: This will likely be reworked a lot as we add capabilities, good for this issue / PR though :)

Comment (rohitsanj) [reply testbook/client.py]: Sure :D

**Comment by rohitsanj** [comment]
Todo:
- Write unit tests
- Add a way to execute multiple cells by passing a list to `execute_cell`

**Comment by rohitsanj** [comment]
I've created a fixture called `notebook` (we should probably change the name of this later)
I've also added a test.

Details of functionality added:
- Execute one or more cells with `TestbookNotebookClient.execute_cell`
- function`cell_output` - get the `outputs` array of a cell.
- function `cell_output_text` - get text output from a cell. 

I will need to create a function that returns the `text/plain` or  `application/json` of `execute_result` as well.

@MSeal  How do you suggest I go about designing an API that can return these values? We could probably pass in a parameter to `cell_outputs` to filter based on `output_type`.

File: testbook/execute.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+@pytest.fixture
+def notebook():
+    @contextmanager
+    def notebook_loader(nb_path, **kwargs):
+        with open(nb_path) as f:
+            nb = nbformat.read(f, as_version=4)
+
+        client = TestbookNotebookClient(nb)
+        with client.setup_kernel(**kwargs):
```
Comment (MSeal) [inline testbook/execute.py]: I've got a PR that will make it so we can control the context and cleanup on user behalf without them needing use `with ..` syntax in their test. For now this is fund but we'll want to come back to it as unless the fixture handles cleanup for us then we might as well just let make this call `with TestbookNotebookClient(nb_path).setup(**kwargs) as nb:` work.

File: testbook/execute.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+from contextlib import contextmanager
+
+import nbformat
+import pytest
+
+from testbook.client import TestbookNotebookClient
+
+
+@pytest.fixture
+def notebook():
```
Comment (MSeal) [inline testbook/execute.py]: Probably better to call this notebook_loader so it's clear it has to be called.

File: testbook/tests/test_execute.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import pytest
+
+from testbook import notebook
+
+
+def test_notebook(notebook):
+    with notebook('testbook/tests/resources/foo.ipynb') as nb:
+        nb.execute_cell(1)
+        assert nb.cell_output_text(1) == 'hello world\n[1, 2, 3]\n'
```
Comment (MSeal) [inline testbook/tests/test_execute.py]: nice!

File: testbook/execute.py (hunk 5 above)
Comment (MSeal) [reply testbook/execute.py]: For now we'll merge and come back to fixtures and naming

**Comment by MSeal** [comment]
Ahh shoot, sorry I missed the WIP tag. That's ok, we can open further PRs as we go since it's still unreleased code that's being molded.

**Pull request merged**